autorest/adal/token.go

```diff
@@ -354,6 +354,7 @@ type ServicePrincipalToken struct {
 	customRefreshFunc TokenRefresh
 	refreshCallbacks  []TokenRefreshCallback
 	// MaxMSIRefreshAttempts is the maximum number of attempts to refresh an MSI token.
+	// Settings this to a value less than 1 will use the default value.
```

Comment:
The other option would be to return an error during token acquisition.  While this approach is more forgiving, it could mask other bugs (e.g. the original issue would have been much harder to diagnose).

Reply:
However the Go idiom is typically to have a meaningful zero-value which usually translates to "give me the default behavior".

Reply:
I'm more inclined to returning the error exactly to avoid masking other bugs, but for now I think it's fine to leave it as is with the comment.